types/project.ts

```diff
@@ -1,10 +1,33 @@
-import type { GCPVendorSettings, AWSVendorSettings, AzureVendorSettings } from "./vendor"
+import type {
+  GCPVendorSettings,
+  AWSVendorSettings,
+  AzureVendorSettings,
+} from './vendor'
 
 export interface Project {
+  project_id?: number
   project_name: string
-  project_id: string
   project_key: string
   project_description: string
-  project_vendor: 'gcp' | 'aws' | 'azure'
-  project_vendor_settings: GCPVendorSettings | AWSVendorSettings | AzureVendorSettings
+  vendor_id: number
+  project_settings:
+    | string
+    | GCPVendorSettings
+    | AWSVendorSettings
+    | AzureVendorSettings
+}
+
+export interface NewProject {
+  project_name: string
+  project_description: string
+  vendor_id: number
+  aws_access_key?: string
+  aws_secret?: string
+  aws_default_zone?: string
+  gcp_access_key?: string
+  gcp_secret?: string
+  gcp_default_zone?: string
+  azure_access_key?: string
+  azure_secret?: string
+  azure_default_zone?: string
```

Comment:
I believe there is room for improvement here rather than the use of optionals. 
call it credentials which should be of type PlatformCredential so a new project can be instantiated with AWSCredential, GCPCredential... based on a mapping. i believe in Typescript, this will perhaps be typed based on a union

Reply:
Updated in the next PR